openedx/core/djangoapps/oauth_dispatch/dot_overrides.py

```diff
@@ -61,6 +88,16 @@ def save_bearer_token(self, token, request, *args, **kwargs):
 
         super(EdxOAuth2Validator, self).save_bearer_token(token, request, *args, **kwargs)
 
+        is_application_restricted = RestrictedApplication.objects.filter(application=request.client).exists()
+
+        if is_application_restricted:
+            # For now, since Open edX doesn't have the protection of scopes, any access token
+            # that was generated for 'restricted applications' (aka external 3rd parties)
+            # must be automatically expired. So let's update
+            # the token dictionary, so set the expires_in field be
+            # on Jan. 1, 1970
+            token['expires_in'] = -time_since_epoch()
```

Comment:
Rather than assuming the `expires_in` value in the database equals the epoch time, can we read the value from the database here and set it according to the database value?